### PR TITLE
Clippy run

### PR DESCRIPTION
**Pull request by leodasvacas**
Fix bunch of lints suggested by clippy and some other refactorings. I'm unsure about removing the return value of some methods in `query.rs` that always returned the same value. Perhaps you meant to change that later? I'm willing to revert any unwanted changes.

### CODE REVIEW

File: src/query.rs [hunk 1]
```diff
@@ -38,20 +35,14 @@ impl Query {
...
-        if self.index <= 0 { return false; }
-
-        match self.query.get(self.index-1) {
-            Some(ch) => {
-                self.index -= 1;
-                self.pos -= if ch.len_utf8() > 1 {2} else {1};
-            }
-            None => {}
+    pub fn backward_char(&mut self) {
+        if let Some(ch) = self.query.get(self.index-1) {
```
Comment (lotabout) [inline src/query.rs:43]: I believe when `self.index` is `0`, `self.index-1` will cause `subtract overflow`.

File: src/query.rs [hunk 2]
```diff
@@ -1,3 +1,4 @@
+#[derive(Default)]
```
Comment (lotabout) [inline src/query.rs:1]: Nice! Learned a new thing :)

**Comment by leodasvacas** [comment]
Sorry about that bug, fixed it and added a test.

**Comment by lotabout** [comment]
Thank you very much for the PR!

**Pull request merged**